src/libraries/System.Private.CoreLib/src/System/Runtime/ExceptionServices/ExceptionHandling.cs

```diff
@@ -48,12 +50,33 @@ public static void SetUnhandledExceptionHandler(Func<Exception, bool> handler)
         /// event and then return.
         ///
         /// It will not raise the the handler registered with <see cref="SetUnhandledExceptionHandler"/>.
+        ///
+        /// This API is thread safe and can be called from multiple threads. However, only one thread
+        /// will trigger the event handlers, while other threads will wait until the event is raised
+        /// and return without raising the event again.
         /// </remarks>
         public static void RaiseAppDomainUnhandledExceptionEvent(object exception)
         {
             ArgumentNullException.ThrowIfNull(exception);
 
-            AppContext.OnUnhandledException(exception);
+            if (Interlocked.CompareExchange(ref s_unhandledExceptionInProgress, -1, 0) == 0)
+            {
+                AppContext.OnUnhandledException(exception);
```

Comment:
I think this logic protection should be moved inside OnUnhandledException. We want to avoid duplicate invocation of unhandled exception handlers when one thread is crashing with regular unhandled exception and another thread is crashing via RaiseAppDomainUnhandledExceptionEvent.

Reply:
Okay. I wanted to put it in there, but assumed there was a reason `RaiseAppDomainUnhandledExceptionEvent` has been suggested. Will move.